src/Model/Message/MessageBagInterface.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PhpLlm\LlmChain\Model\Message;
+
+/**
+ * @template-extends \ArrayAccess<int, MessageInterface>
+ * @template-extends \IteratorAggregate<int, MessageInterface>
+ */
+interface MessageBagInterface extends \IteratorAggregate, \ArrayAccess, \Serializable, \Countable, \JsonSerializable
```

Comment:
```php
extends \IteratorAggregate, \ArrayAccess, \Serializable, \Countable,
```
are all from `\ArrayObject`, not sure we need all of them

Reply:
alternative would be to decouple from that stuff. the only thing we get from that currently is the `$messages[]`

not using `count()` nor iterating over it. array keys are not used as well. at least in my use cases the only thing would be to add an `$messages->add()` method

Reply:
WDYT?

Reply:
@DZunke any concerns from your end with the breaking change to replace `$messages[] = $message` by `$messages->add($message)`

Reply:
Separate PR created:
* https://github.com/php-llm/llm-chain/pull/174